packages/reaction-core/server/methods/shop.js

```diff
@@ -363,6 +363,35 @@ Meteor.methods({
   },
 
   /**
+   * shop/createTag
+   * @summary creates new tag
+   * @param {String} tagName - new tag name
+   * @param {Boolean} isTopLevel - if true -- new tag will be created on top of
+   * tags tree
+   * @return {String} with created tag _id
+   */
+  "shop/createTag": function (tagName, isTopLevel) {
+    check(tagName, String);
+    check(isTopLevel, Boolean);
+
+    // must have 'core' permissions
+    if (!ReactionCore.hasPermission("core")) {
+      throw new Meteor.Error(403, "Access Denied");
+    }
+
+    // we assume what tag is not exist.
+    const tag = {
```

Comment:
I assume you meant "we assume this tag does not exist" and why do we assume that? Wouldn't a simple check for an existing tag be prudent?